packages/gguf/scripts/generate-llm.ts

```diff
@@ -0,0 +1,179 @@
+/**
+ * Script for generating llm.ts
+ * The source data is taken from llama.cpp
+ */
+
+import { writeFileSync } from "node:fs";
+
+const SOURCE_CPP_URL = "https://raw.githubusercontent.com/ggerganov/llama.cpp/master/llama.cpp";
+const DEST_FILE_PATH = "./src/transformer-llm.ts";
+const DEST_COMMON_SOURCE = `
+type Attention<TArchitecture extends string> =
+	& { [K in \`\${TArchitecture}.attention.head_count\`]: number }
+	& { [K in \`\${TArchitecture}.attention.head_count_kv\`]: number }
+	& { [K in \`\${TArchitecture}.attention.layer_norm_epsilon\`]: number }
+	& { [K in \`\${TArchitecture}.attention.layer_norm_rms_epsilon\`]: number }
+	& { [K in \`\${TArchitecture}.attention.alibi_bias_max\`]: number }
+	& { [K in \`\${TArchitecture}.attention.clip_kqv\`]: number }
+	& { [K in \`\${TArchitecture}.attention.use_norm\`]: number };
+
+type Rope<TArchitecture extends LLMArchitecture> =
+	& { [K in \`\${TArchitecture}.rope.dimension_count\`]: number }
+	& { [K in \`\${TArchitecture}.rope.freq_base\`]: number }
+	& { [K in \`\${TArchitecture}.rope.scale\`]: number }
+	& { [K in \`\${TArchitecture}.rope.scale_linear\`]: number };
+
+type MOE<TArchitecture extends LLMArchitecture> =
+	& { [K in \`\${TArchitecture}.expert_count\`]: number }
+	& { [K in \`\${TArchitecture}.expert_used_count\`]: number };
+
+export type TransformerLLMArchitecture = LLMArchitecture; // type alias
+export type TransformerLLMBase<TArchitecture extends LLMArchitecture> = ModelBase<TArchitecture>
+	& MOE<TArchitecture>
+	& Attention<TArchitecture>
+	& Rope<TArchitecture>;
+`;
+
+const KV_TYPE = {
+	LLM_KV_ATTENTION_LAYERNORM_RMS_EPS: "number",
+	LLM_KV_ATTENTION_LAYERNORM_EPS: "number",
+	LLM_KV_ATTENTION_CAUSAL: "boolean",
+	LLM_KV_TOKENIZER_TOKEN_TYPE_COUNT: "number",
+	LLM_KV_POOLING_TYPE: "number", // TODO: this is an enum
+	LLM_KV_ATTENTION_CLAMP_KQV: "number",
+	LLM_KV_ATTENTION_MAX_ALIBI_BIAS: "number",
+	LLM_KV_SSM_CONV_KERNEL: "number",
+	LLM_KV_SSM_INNER_SIZE: "number",
+	LLM_KV_SSM_STATE_SIZE: "number",
+	LLM_KV_SSM_TIME_STEP_RANK: "number",
+	LLM_KV_LOGIT_SCALE: "number",
+};
+
+interface Arch {
+	cppConst: string,      // for example: "LLM_ARCH_LLAMA"
+	name: string,          // for example: "llama"
+	tsName: string,        // for example: "ArchLlama"
+	tensorNames: string[], // for example: "token_embd"
+	hparams: string[],
+};
+
+async function main() {
+	const res = await fetch(SOURCE_CPP_URL);
+	const cppSource = await res.text();
+
+	/////////////////////////////////////
+	// extract list of all architectures
+	const archList: Arch[] = [];
+	const matchedArchList = cppSource.match(/LLM_ARCH_NAMES = ([^;]+)/)!![1].split("\n");
+	for (const line of matchedArchList) {
+		const matched = line.match(/(LLM_ARCH_[A-Z0-9_]+),\s+"([^"]+)"/);
```

Comment:
```suggestion
		const matched = line.match(/(LLM_ARCH_[A-Z0-9_]+),\s+"(.+?)"/);
```

I think you can use the [greedy/lazy regex quantifiers](https://javascript.info/regexp-greedy-and-lazy). Let me know if this change works. If so, uou can replace all instances of `"([^"]+)"` with `"(.+?)"`

Reply:
Thanks for pointing me in the correct direction. I replaced all instances of `"([^"]+)"` with `"(.+?)"`. However, the same technique does not work with `([^;]+)` ==> `(.+?);` (matches `ABC;` for example). Please let me know if you have any idea. Thanks.

Reply:
`([^;]+)` is fine as it is